lib/galaxy/webapps/galaxy/services/history_contents.py

```diff
@@ -125,7 +124,7 @@
 
 log = logging.getLogger(__name__)
 
-DatasetDetailsType = Union[Set[EncodedDatabaseIdField], Literal["all"]]
+DatasetDetailsType = Union[List[EncodedDatabaseIdField], Literal["all"]]
```

Comment:
This should be a DecodedDatabaseIdField IMO, but we can refactor that later.

Reply:
Yeah, I wanted to keep the changes minimal for the fix, but probably a good idea to start refactoring that as we touch the code. I'll see if I can refactor it here.

Reply:
Hmm, too many changes... I think I'll open a different PR targeting `dev` instead.

Reply:
xref https://github.com/galaxyproject/galaxy/pull/14392